service/controller/clusterapi/v29/resource/clusterazs/create.go

```diff
@@ -112,15 +119,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return microerror.Mask(err)
 		}
 
-		ccAZs := mapAZSubnetsToControllerContextTypes(azs)
+		ccSpecAZs := mapAZSubnetsToControllerContextTypes(azs)
 
 		r.logger.LogCtx(ctx, "level", "debug", "message", azSubnetsToString(azs))
 
-		// TODO(marcel):
-		// Also maintain cc.Spec.TenantCluster.AvailabilityZones here to allow
-		// detection.ShouldUpdate() work. See also TODO in
-		// ensureAZsAreAssignedWithSubnet() below.
-		cc.Status.TenantCluster.AvailabilityZones = ccAZs
+		// Add the desired AZ state to the controllercontext spec.
+		cc.Spec.TenantCluster.AvailabilityZones = ccSpecAZs
```

Comment:
Use Spec instead here for the desired state.

Reply:
Why don't we simplify it here as with the status. So just calling `newAZSpec()`?